### PR TITLE
[sovereign] indexValidatorsRatingIfNeeded

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
-  index validator ratings at each `CommitBlock` if outport driver is attached
- `SaveValidatorsPubKeys` at each epoch change event detected
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/block/sovereignChainBlock.go [hunk 1]
```diff
@@ -1515,15 +1515,28 @@ func (scbp *sovereignChainBlockProcessor) CommitBlock(headerHandler data.HeaderH
...
 		return err
 	}
+
 	// TODO: MX-15748 Analyse if !check.IfNil(lastMetaBlock) && lastMetaBlock.IsStartOfEpochBlock() from metablock.go
 	err = scbp.commonHeaderAndBodyCommit(headerHandler, body, headerHash, []data.HeaderHandler{lastSelfNotarizedHeader}, [][]byte{lastSelfNotarizedHeaderHash})
 	if err != nil {
 		return err
 	}
 
+	scbp.indexValidatorsRatingIfNeeded(headerHandler)
```
Comment (axenteoctavian) [inline process/block/sovereignChainBlock.go:1525]: I believe this should happen only on epoch start block

Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:1525]: No, these are the ratings, those are updated at each round. It's the same as it is for meta

Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:1525]: Actually, you are right.

Comment (sasurobert) [reply process/block/sovereignChainBlock.go:1525]: it is only on epochStartBlock.

Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:1525]: Fixed it

File: factory/status/common.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	nodeData "github.com/multiversx/mx-chain-core-go/data"
+	outportCore "github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/epochStart"
+	"github.com/multiversx/mx-chain-go/epochStart/notifier"
+	"github.com/multiversx/mx-chain-go/outport"
+	"github.com/multiversx/mx-chain-go/sharding/nodesCoordinator"
+)
+
+func CreateSaveValidatorsPubKeysEventHandler(
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply factory/status/common.go:13]: Right, added comm

**Review by axenteoctavian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**